.travis.yml

```diff
@@ -10,6 +10,24 @@ install:
   - pip install --upgrade pip
   - pip install .
   - pip install -r dev-requirements.txt
+  - sudo apt-get update
+  # We do this conditionally because it saves us some downloading if the
+  # version is the same.
+  - if [[ "$TRAVIS_PYTHON_VERSION" == "2.7" ]]; then
+      wget https://repo.continuum.io/miniconda/Miniconda2-latest-Linux-x86_64.sh -O miniconda.sh;
+    else
+      wget https://repo.continuum.io/miniconda/Miniconda3-latest-Linux-x86_64.sh -O miniconda.sh;
+    fi
+  - bash miniconda.sh -b -p $HOME/miniconda
+  - export PATH="$PATH:$HOME/miniconda/bin"
```

Comment:
Shouldn't we put Conda at the front of the PATH instead of the end so that we find it first? This might change which`python` we use but it's probably for the best. In that case we should probably install Conda before the pip requirements too and use the pip in Conda.

Reply:
Yeah given MLflow's focus on conda envs it'd be good for users to be able to run tests within a conda environment.

It actually looks conda + tox isn't well-supported, since tox runs tests in a virtualenv & creating a virtualenv within a conda env isn't yet well supported (see [this thread](https://github.com/tox-dev/tox/issues/273) on tox / conda integration). However installing virtualenv via conda before running `tox` seems to work, so I've done that for now.

If we're ok moving away from tox & virtualenv I think we can mimic Keras's [.travis.yml](https://github.com/keras-team/keras/blob/master/.travis.yml), which just uses conda & pytest.